modules/http/src/main/scala/ch/epfl/bluebrain/nexus/service/http/RejectionHandling.scala

```diff
@@ -0,0 +1,239 @@
+package ch.epfl.bluebrain.nexus.service.http
+
+import akka.http.scaladsl.marshalling.{ToEntityMarshaller, ToResponseMarshallable}
+import akka.http.scaladsl.model.EntityStreamSizeException
+import akka.http.scaladsl.model.StatusCodes._
+import akka.http.scaladsl.model.headers._
+import akka.http.scaladsl.server.AuthenticationFailedRejection.{CredentialsMissing, CredentialsRejected}
+import akka.http.scaladsl.server.Directives.{complete, extractMaterializer, extractRequest}
+import akka.http.scaladsl.server._
+import io.circe.generic.semiauto._
+import io.circe.syntax._
+import io.circe.{Encoder, Json}
+
+import scala.reflect.ClassTag
+
+/**
+  * A generic rejection handling that generates json responses and consumes the request.
+  */
+object RejectionHandling {
+// $COVERAGE-OFF$
+
+  final val errorContextIri = "https://bluebrain.github.io/nexus/contexts/error.json"
+
+  private final case class Error(`@type`: String, reason: String)
+  private object Error {
+    final implicit val genericEncoder: Encoder[Error] =
+      deriveEncoder[Error].mapJson { json =>
+        val context = Json.obj("@context" -> Json.fromString(errorContextIri))
+        json deepMerge context
+      }
+  }
+
+  /**
+    * Discards the request entity bytes and completes the request with the argument.
+    *
+    * @param m a value to be marshalled into an HttpResponse
+    */
+  final def rejectRequestEntityAndComplete(m: ⇒ ToResponseMarshallable): Route = {
+    extractRequest { request ⇒
```

Comment:
Probably copy/pasted from Akka but I'd rather not have unicode arrows in our source files.